vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.grid.dataview;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.ComponentUtil;
+import com.vaadin.flow.component.grid.Grid;
+import com.vaadin.flow.data.provider.DataCommunicator;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.data.provider.SizeChangeEvent;
+import com.vaadin.flow.shared.Registration;
+
+public class GridDataViewImpl<T> implements GridDataView<T> {
+
+    private DataCommunicator<T> dataCommunicator;
+    private Grid<T> grid;
+
+    public GridDataViewImpl(DataCommunicator<T> dataCommunicator,
+            Grid<T> grid) {
+        this.dataCommunicator = dataCommunicator;
+        this.grid = grid;
+    }
+
+    @Override
+    public Stream<T> getItems() {
+        return dataCommunicator.getDataProvider()
+                .fetch(dataCommunicator.buildQuery(0, Integer.MAX_VALUE));
```

Comment:
Not sure if this should use the `getSize()` as the limit instead of `Integer.MAX_VALUE`.
The lazy case could then override this as it probably should do anyway for the javadocs.